cg/meta/demultiplex/status_db_storage_functions.py

```diff
@@ -55,6 +55,24 @@ def store_flow_cell_data_in_status_db(
     store.session.commit()
 
 
+def check_if_samples_have_files(flow_cell: FlowCellDirectoryData) -> None:
+    """Check if all samples have already all the fastq files in the demultiplex directory.
+    Raises: MissingFilesError
+        When one of the samples does not have enough fastq files in the flow cell
+    """
+    sample_ids: List[str] = get_sample_internal_ids_from_sample_sheet(
+        sample_sheet_path=flow_cell.get_sample_sheet_path_hk(),
+        flow_cell_sample_type=flow_cell.sample_type,
+    )
+    for sample_id in sample_ids:
+        fastq_files: Optional[List[Path]] = get_sample_fastqs_from_flow_cell(
+            flow_cell_directory=flow_cell.path, sample_internal_id=sample_id
+        )
+        if not fastq_files or len(fastq_files) != NUMBER_FASTQ_FILES_PER_SAMPLE:
+            error_msg: str = f"Sample {sample_id} does not have the correct number of fastq files"
+            raise MissingFilesError(error_msg)
```

Comment:
Ok great info, thanks!

Reply:
Is there any way to distinguish the negative controls from the actual samples in the sample sheet? Maybe the current validation is fine, but if we want to it should be possible to check that all the expected files actually are present

Reply:
There is a field about that in status_db but I still think this would not be the best way to go. If a customer sends us a terrible sample it will probably fail our QC threshold and not have its FASTQ files added. That should not stop the processing of the rest of the samples on the flow cell. To me the two options are: either tying together the housekeeper and status_db function (plz no) or make the check more general and check if we find any FASTQ files